rl4co/models/nn/attention.py

```diff
@@ -129,97 +130,104 @@ def forward(self, x, key_padding_mask=None):
         self.sdp_fn = sdp_fn
-
-    def forward(self, query, key, value, logit_key, mask, softmax_temp=None):
+        self.check_nan = check_nan
+
+        # Check unused kwargs
+        if unused_kwargs:
+            log.warning(f"Unused kwargs: {unused_kwargs}")
+
+    def forward(self, query, key, value, logit_key, attn_mask=None):
+        """Compute attention logits given query, key, value, logit key and attention mask.
+
+        Args:
+            query: query tensor of shape [B, ..., L, E]
+            key: key tensor of shape [B, ..., S, E]
+            value: value tensor of shape [B, ..., S, E]
+            logit_key: logit key tensor of shape [B, ..., S, E]
+            attn_mask: attention mask tensor of shape [B, ..., S]. Note that `True` means that the value _should_ take part in attention
+                as described in the [PyTorch Documentation](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html)
+        """
         # Compute inner multi-head attention with no projections.
-        heads = self._inner_mha(query, key, value, mask)
+        heads = self._inner_mha(query, key, value, attn_mask)
         glimpse = self.project_out(heads)
 
         # Batch matrix multiplication to compute logits (batch_size, num_steps, graph_size)
         # bmm is slightly faster than einsum and matmul
-        logits = (
-            torch.bmm(glimpse, logit_key.squeeze(1).transpose(-2, -1))
-            / math.sqrt(glimpse.size(-1))
-        ).squeeze(1)
-
-        # From the logits compute the probabilities by clipping, masking and softmax
-        if self.tanh_clipping > 0:
-            logits = torch.tanh(logits) * self.tanh_clipping
-
-        if self.mask_logits:
-            logits[mask] = float("-inf")
-
-        # Normalize with softmax and apply temperature
-        if self.normalize:
-            softmax_temp = softmax_temp if softmax_temp is not None else self.softmax_temp
-            logits = torch.log_softmax(logits / softmax_temp, dim=-1)
+        logits = (torch.bmm(glimpse, logit_key.squeeze(-2).transpose(-2, -1))).squeeze(
+            -2
+        ) / math.sqrt(glimpse.size(-1))
 
-        assert not torch.isnan(logits).any(), "Logits contain NaNs"
+        if self.check_nan:
+            assert not torch.isnan(logits).any(), "Logits contain NaNs"
 
         return logits
 
-    def _inner_mha(self, query, key, value, mask):
+    def _inner_mha(self, query, key, value, attn_mask):
         q = self._make_heads(query)
         k = self._make_heads(key)
         v = self._make_heads(value)
-
         if self.mask_inner:
-            # need to invert mask: (N L S) -> (N 1 L S)
+            # make mask the same number of dimensions as q
             attn_mask = (
-                ~mask.unsqueeze(1) if mask.ndim == 3 else ~mask.unsqueeze(1).unsqueeze(2)
+                attn_mask.unsqueeze(1)
+                if attn_mask.ndim == 3
```

Comment:
Since we are considering multi batch dimensions in this version, should this be modified?

Reply:
Actually for the `batchify` refactoring we considered it might take a bit more time and effort (we planned to support _any_ leading batch dimension for all operations, which right now is not supported since we do reshaping via `batchify` and `unbatchify`).
I think still this is a bit low in the priority list since it turns out many libraries do not consider multiple batch dimensions - PyTorch's own [MultiheadAttention](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html) does not, for instance